recordedit/recordEdit.app.js

```diff
@@ -265,37 +265,43 @@
 
                         // populate defaults
                         angular.forEach($rootScope.reference.columns, function(column) {
-                            if (!column.getInputDisabled(context.appContext)) {
-                                // if column.default == undefined, the second condition would be true so we need to check if column.default is defined
-                                // only want to set values in the input fields so make sure it isn't a function
-                                // check the recordEditModel to make sure a value wasn't already set based on the prefill condition
-                                if (column.default !== undefined && typeof column.default !== "function" && !recordEditModel.rows[0][column.name]) {
-                                    if (column.type.name === 'timestamp' || column.type.name === 'timestamptz') {
-                                        if (column.default !== null) {
-                                            var ts = moment(column.default);
-                                            recordEditModel.rows[0][column.name] = {
-                                                date: ts.format('YYYY-MM-DD'),
-                                                time: ts.format('hh:mm:ss'),
-                                                meridiem: ts.format('A')
-                                            };
-                                        } else {
-                                            recordEditModel.rows[0][column.name] = {
-                                                date: null,
-                                                time: null,
-                                                meridiem: 'AM'
-                                            };
-                                        }
-                                    } else if (column.isAsset) {
+                            // if column.default == undefined, the second condition would be true so we need to check both if column.default is defined and it's not a function
+                            // only want to set values in the input fields so make sure it isn't a function
+                            // check the recordEditModel to make sure a value wasn't already set based on the prefill condition
+                            if (column.default !== undefined && typeof column.default !== "function" && !recordEditModel.rows[0][column.name]) {
```

Comment:
That check is being made because you could have come from an `Add link` for a related table. This would populate `recordEditModel.rows[x][column.name]` with a value representing a link to the entity that that related table was associated with. To the best of my knowledge, a foreign key relationship should never be represented by a `boolean` value. So I think it is safe to assume if it's already set, it won't be a boolean value.